### PR TITLE
mute/unmute chat

### PR DESCRIPTION
**Pull request by renxinhe**
Closes Schmavery/facebook-chat-api#294.
Implemented mute/unmute API, `muteChat.js`, based on `markAsRead.js`.

### CODE REVIEW

File: src/muteChat.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+module.exports = function(defaultFuncs, api, ctx) {
+  // muteSecond: -1=permanent mute, 0=unmute, 60=one minute, 3600=one hour, etc.
+  return function muteChat(threadID, muteSecond, callback) {
+    if(!callback) {
+      callback = function() {};
+    }
+
+    var form = {};
+    form["thread_fbid"] = threadID;
+    form["mute_settings"] = muteSecond;
```
Comment (bsansouci) [inline src/muteChat.js]: Can you add those directly in the object
```js
var form = {
  thread_fbid: threadID,
  mute_settings: muteSeconds,
}
```

File: src/muteChat.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(defaultFuncs, api, ctx) {
+  // muteSecond: -1=permanent mute, 0=unmute, 60=one minute, 3600=one hour, etc.
+  return function muteChat(threadID, muteSecond, callback) {
```
Comment (bsansouci) [inline src/muteChat.js]: Can you add an s to `muteSecond` so it's `muteSeconds`?

**Comment by bsansouci** [comment]
This looks great! Minor fixes and then we can merge :D

**Comment by bsansouci** [comment]
Oh two more things that I just realized. We call chats `thread`s. Can you update the name to be `muteThread`?
Also you need to add `"muteThread"` here https://github.com/Schmavery/facebook-chat-api/blob/fddb01ffaf0e9119f6b44f78c2ed38e892bdb324/index.js#L88

**Comment by Schmavery** [comment]
Thanks for working on this.  If you could double check to make sure your example code does what it's meant to, that'd be awesome.

**Comment by GAMELASTER** [comment]
Well, its okay! But is good to mention the status if the user is muted contains a `message` variable on `listen`. And I'm not sure what is the variable name + I think it is contains a time to when is it muted

**Comment by renxinhe** [comment]
You welcome! I ran the example on a node terminal, and a friend got muted for a minute after sending me a message. 
Messenger.com correctly showed "Muted until 1:02 pm"; I received the message around 1:01 pm.

**Comment by GAMELASTER** [comment]
@renxinhe I'm sorry, the information about mute contains threadlist_info.php request:
image

**Comment by renxinhe** [comment]
Sorry @GAMELASTER , what's the issue here? I just looked at `api.listen`, and I noticed that certain message types like `file` and `photo` don't have a `threadID`. Is that what you were saying that I should handle?

**Comment by GAMELASTER** [comment]
Ah, sorry! I just wanted to write into a documentation into a [getThreadList](https://github.com/Schmavery/facebook-chat-api/blob/master/DOCS.md#getThreadList) how to get the mute state for thread, it is mute_settings variable on thread list info. But well, isn't in good format, maybe we can add some parsing for just return `mute_settings: true`.

**Comment by renxinhe** [comment]
Oh I see. I can make an `api.getMuteStatus` function in another pull request if necessary.

**Comment by Schmavery** [comment]
This would be added to the getThreadList response, not a new function, but sure, this can be added later.  Thanks for the investigation @GAMELASTER.

**Comment by renxinhe** [comment]
Cool! Guess my job is done. It's curious to see that Facebook doesn't return a `threadID` for images and files. One'd assume everything is associated with a message thread.

**Comment by bsansouci** [comment]
Awesome! Thanks a lot @renxinhe. Can you follow up with a PR to add `muteStatus` onto the thread object ([here](https://github.com/Schmavery/facebook-chat-api/blob/def4518900c2acf24d2e3ac0c84ce7ecd95fa859/utils.js)) whenever you get some time.

**Comment by renxinhe** [comment]
Sure. Just add it to `function formatThread (data)` correct?

**Comment by bsansouci** [comment]
Yes, sorry didn't copy paste the URL properly.

**Comment by renxinhe** [comment]
@bsansouci The [thread object](https://github.com/Schmavery/facebook-chat-api/blob/master/utils.js#L621) already has a `muteSettings` attribute.

**Comment by GAMELASTER** [comment]
@renxinhe well, yeah, it is, but it needs a parsing, look into mine image, its in format:
mute_settings:
  THREADID@facebook.com: CONSTANT
It will be nice to mute_settings just contains a value. I think enough will be just get a mute_settings first children variable value)

**Comment by bsansouci** [comment]
@renxinhe What @GAMELASTER said ^. Also documentation is needed to explain `-1` and any other value :)

**Pull request merged**